extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/telemetry/MicrometerWebSocketEndpoint.java

```diff
@@ -0,0 +1,59 @@
+package io.quarkus.websockets.next.runtime.telemetry;
+
+import io.micrometer.core.instrument.Counter;
+import io.quarkus.websockets.next.runtime.WebSocketEndpoint;
+import io.vertx.core.Future;
+import io.vertx.core.buffer.Buffer;
+
+final class MicrometerWebSocketEndpoint extends AbstractWebSocketEndpointWrapper {
+
+    private final Counter onMessageReceivedCounter;
+    private final Counter onMessageReceivedBytesCounter;
+
+    MicrometerWebSocketEndpoint(WebSocketEndpoint delegate, Counter onMessageReceivedCounter,
+            Counter onMessageReceivedBytesCounter) {
+        super(delegate);
+        this.onMessageReceivedCounter = onMessageReceivedCounter;
+        this.onMessageReceivedBytesCounter = onMessageReceivedBytesCounter;
+    }
+
+    @Override
+    public Future<Void> onTextMessage(Object message) {
+        addMetricsIfMessageIsString(message);
+        return delegate.onTextMessage(message);
+    }
+
+    @Override
+    public Future<Void> onBinaryMessage(Object message) {
+        addMetricsIfMessageIsBuffer(message);
+        return delegate.onBinaryMessage(message);
+    }
+
+    @Override
+    public Object decodeTextMultiItem(Object message) {
+        addMetricsIfMessageIsString(message);
+        return delegate.decodeTextMultiItem(message);
+    }
+
+    @Override
+    public Object decodeBinaryMultiItem(Object message) {
+        addMetricsIfMessageIsBuffer(message);
+        return delegate.decodeBinaryMultiItem(message);
+    }
+
+    private void addMetricsIfMessageIsString(Object message) {
+        if (message instanceof String stringMessage) {
+            onMessageReceivedCounter.increment();
+            double bytesNum = stringMessage.getBytes().length;
```

Comment:
`double`?

Reply:
No, I was just wondering why counter expects a double?

Reply:
AFAICT from `/q/metrics` all the metrics has floating point anyway, even if you only use `increment` method and increment by one